src/behavior-property-observer.js

```diff
@@ -92,3 +117,51 @@ export class BehaviorPropertyObserver {
     this.removeSubscriber(context, callable);
   }
 }
+
+const numCons = Number;
+const dateCons = Date;
+const _isFinite = isFinite;
+const _isNaN = isNaN;
+
+export const coerces = {
+  none(a) {
+    return a;
+  },
+  number(a) {
+    var val = numCons(a);
+    return !_isNaN(val) && _isFinite(val) ? val : 0;
+  },
+  string(a) {
+    return '' + a;
+  },
+  boolean(a) {
+    return !!a;
+  },
+  date(a) {
+    return new dateCons(a);
+  }
+};
+
+/**@type {Map<Function, string>} */
+export const classCoerceMap = new Map([
```

Comment:
I think that instead of the two lines above You can write smth like:
`export const classCoerceMap: Map<Function, string> = new Map([`
or even better (more precise type):
`export const classCoerceMap: Map<(strValue: string) => any, string> = new Map([`
As it would be correct way to write type information for TypeScript variables/constants, and at the moment some Babel plugin is used to extract this information to generate API docs and definitions files for TS.
My only concern is that maybe the Babel plugin that extracts TS types doesn't understand so complex TS type expressions as i wrote in the second example